File: Moving Zeros/moveZeros_v1.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        i = 0
        while i < n:
            if nums[i] == 0:
                nums.append(0) # add zero to end
                nums.pop(i) # remove zero from curr position
                n-= 1 # reduce 
            else:
                i += 1 # only increase if no zero is removed in case of trailing zeros


# same concept as the v1 in C++

File: Moving Zeros/moveZeros_v1.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();

        for ( int i = 0; i < n;){
            if ( nums[i] == 0){
                nums.push_back(0); // push 0 at end
                nums.erase(nums.begin() + i); // erase at location begin() + index
                n--; // adjust the length after moving zero to end
            } else {
                i++; // increase i only if its not zero
            }
        }
    }
};
